### DISCUSSION TITLE
relate bought digital products to customers

### DISCUSSION THREAD

**Discussion by PeterAlfonsLoch** [pocketbase/pocketbase#928] [Q&A]
I am making a platform for users to purchase online courses and want to limit access to the users who bought it. How can I relate what the user has paid for? They may buy multiple products. I am used to relational databases like MYSql and I am unsure on how to implement it via pocketbase.

**Comment by benallfree** [comment]
Have you considered adding a json field to the user collection? A json array of purchased products ids. I’d do this instead of a junction table.

**Reply by PeterAlfonsLoch** [reply]
I did consider it, but If I can avoid having to rewrite json i also get rid of a potential place where errors can occur.

**Comment by ganigeorgiev** [comment]
@PeterAlfonsLoch

>  I am used to relational databases like MYSql and I am unsure on how to implement it via pocketbase.

PocketBase data structure is not much different than the other traditional relation databases. Under the hood, collections are translated to plain SQLite tables.

---

> How can I relate what the user has paid for?

You can use the API rules for that. They act both as admin level access and filter controls. There will be more documentation on this in the next v0.8.0 release, but you could find some info in https://pocketbase.io/docs/manage-collections/#collection-api-rules.

There are several ways to handle this depending on how you want to operate with your data, but as the other users suggested, you can create a junction collection, eg. `userCourses` with "user" and "course" `relation` fields and use that for association.

Then for example in your `courses` collection  you can have a list and view filter rules like:

```js
// for PocketBase v0.8.x
@collection.userCourses.course = id && @collection.userCourses.user = @request.auth.id

// for PocketBase v0.7.x - v1 if in the junction collection you use the `user` relation field
@collection.userCourses.course = id && @collection.userCourses.user = @request.user.id

// for PocketBase v0.7.x - v2 if in the junction collection you use relation to the profiles collection
@collection.userCourses.course = id && @collection.userCourses.user = @request.user.profile.id
```

But at the end it all depends on how you decide to structure your data and to what you want to give access.

**Comment by PeterAlfonsLoch** [comment]
I have found a solution, since i can add multiple relations to one client i can just realte the course to them and then check at the user if they have a relation to a course.
